src/components/common/ColonyDecisions/DecisionDialog/DecisionDialog.tsx

```diff
@@ -49,27 +49,27 @@ const validationSchema = object()
 export type DecisionDialogValues = InferType<typeof validationSchema>;
 
 export interface DecisionDialogProps extends DialogProps {
+  decision?: Decision;
   ethDomainId?: number;
-  handleSubmit?: (values: DecisionDialogValues) => void;
 }
 
 const DecisionDialog = ({
   cancel,
   close,
+  decision,
   ethDomainId,
-  handleSubmit,
 }: DecisionDialogProps) => {
   const { user } = useAppContext();
   const { pathname } = useLocation();
   const { editor } = useRichTextEditor();
+  const dispatch = useDispatch();
   const navigate = useNavigate();
   const domainId = getDomainId(ethDomainId);
   const walletAddress = user?.walletAddress || '';
-  const draftDecision = getDecisionFromLocalStorage(walletAddress);
-  const content = draftDecision?.description;
+  const content = decision?.description;
 
   const handleSubmitDialog = (values: DecisionDialogValues) => {
-    handleSubmit?.(values);
+    dispatch({ type: ActionTypes.DECISION_DRAFT_CREATED, payload: values });
     setDecisionToLocalStorage(values, walletAddress);
     if (!pathname.includes(DECISIONS_PREVIEW)) {
       navigate(`${pathname}${DECISIONS_PREVIEW}`);
```

Comment:
If we're setting the decision inside redux, shouldn't the local storage step moved inside a saga then ?